OMCompiler/Compiler/Script/CevalScriptBackend.mo

```diff
@@ -3387,6 +3391,161 @@ algorithm
   end matchcontinue;
 end runFrontEndWork;
 
+
+/* -------------------------------------------------------------------
+                         RUN NEW FRONTEND
+   ------------------------------------------------------------------- */
+public function runNewFrontEnd
+  "author: kabdelhak 2020-03
+   Only run the new frontend pipeline and return the flat model instead of
+   DAE representation. Used for new backend and should replace runFrontEnd
+   at some point."
+  input FCore.Cache cache;
+  input FCore.Graph env;
+  input Absyn.Path className;
+  output Option<FlatModel> optFlatModel = NONE();
+  output Option<FunctionTree> optFuncs = NONE();
+  input Boolean relaxedFrontEnd "Do not check for illegal simulation models, so we allow instantation of packages, etc";
+protected
+  FlatModel flatModel;
+  FunctionTree funcs;
+  Boolean b;
+algorithm
+  // add program to the cache so it can be used to lookup modelica://
+  // URIs in external functions IncludeDirectory/LibraryDirectory
+  FlagsUtil.setConfigBool(Flags.BUILDING_MODEL, true);
+  try
+    b := runFrontEndLoadProgram(className);
+    true := b;
+    if Flags.isSet(Flags.GC_PROF) then
+      print(GC.profStatsStr(GC.getProfStats(), head="GC stats before front-end:") + "\n");
+    end if;
+    ExecStat.execStat("FrontEnd - loaded program");
+    (flatModel, funcs) := runNewFrontEndWork(cache,env,className,relaxedFrontEnd,Error.getNumErrorMessages());
+    if Flags.isSet(Flags.GC_PROF) then
+      print(GC.profStatsStr(GC.getProfStats(), head="GC stats after front-end:") + "\n");
+    end if;
+    ExecStat.execStat("FrontEnd - DAE generated");
+    optFlatModel := SOME(flatModel);
+    optFuncs := SOME(funcs);
+  else
+    // Return optFlatModel=NONE(); and optFuncs=NONE();
+  end try;
+  FlagsUtil.setConfigBool(Flags.BUILDING_MODEL, false);
+end runNewFrontEnd;
+
+protected function runNewFrontEndWork
+  input FCore.Cache inCache;
+  input FCore.Graph inEnv;
+  input Absyn.Path className;
+  input Boolean relaxedFrontEnd "Do not check for illegal simulation models, so we allow instantation of packages, etc";
+  input Integer numError;
+  output FlatModel flatModel;
+  output FunctionTree funcs;
+algorithm
+  (flatModel, funcs) := matchcontinue (inCache,inEnv,className)
```

Comment:
If possible rewrite this so that we can avoid using matchcontinue. We should really discontinue matchcontinue!

Reply:
Yes, this is just from the original function i copied. Unfortunately i don't know enough about all the stuff going on here to feel comfortable enough to just change it, but i can try!

Reply:
It is basically a match with backtracking (I think it was added due to the need to support some stuff with the old RML implementation)

Reply:
I know what matchcontinue is, but i don't know how to replace it consistently with match, such that it has the same behavior.

Reply:
I see sorry! My bad